Sources/Spezi/SharedRepository/Builtin/ValueRepository.swift

```diff
@@ -0,0 +1,36 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+
+/// A ``ValueRepository`` that allows to store any ``KnowledgeSource``s.
+public typealias UniversalValueRepository = ValueRepository<Any>
+
+
+/// A ``SharedRepository`` implementation that itself is a value type.
+///
+/// - Note: The ``ValueRepository`` is considered `Sendable` if the ``RepositoryAnchor`` conforms to `Sendable`.
+public struct ValueRepository<Anchor>: SharedRepository, BuiltinRepository {
```

Comment:
I don't see any usage of the `ValueRepository`, I suspect this is more meant for other modules to use? We might want to either add some article in the DocC docs on how to use the shared repository mechanism (IMO preferred) or at least add some comment here demonstrating this usage so developers have some examples how to use these mechanisms.

Reply:
This is something I will use e.g. in `SpeziAccount`. A `ValueRepository` (having Value semantics) is better suited for controlling write access (through property mutability, similar to a regular Array or Dictionary). As this is probably a typical suite case I put that implementation here.

Reply:
Sounds good 🚀

Reply:
I added a Shared Repository article to DoCC and also made sure that all types related to Shared Repository are grouped accordingly.